audit/syslogtls.go

```diff
@@ -67,6 +59,31 @@ func (s *SyslogTLS) Shutdown(ctx context.Context) error {
 	return errs.ErrorOrNil()
 }
 
+// getCertPool returns a x509.CertPool containing the cert(s)
+// from `cert`, which can be a path to a .pem or .crt file,
+// or a base64 encoded cert.
+func getCertPool(cert string) *x509.CertPool {
```

Comment:
Is it safe to just return `nil` on error? Shouldn't we return the error itself and manage it as part of `NewSyslogTLSTarget`?

Reply:
Thanks @mgdelacroix 

The underlying Go API's for creating the pool have no specific error conditions so I cannot add any context other than worked or didn't.  Also, the TLS config expects a nil cert pool if one cannot be provided and it will try to connect without it so any error returned would be ignored.

I guess the underlying question is whether we should allow an invalid cert entry to still connect.  I could add a log warning about the invalid cert so it doesn't silently fail.  I'm not so sure about not having auditing for that server instance when it could have connected.  If it cannot connect because it really does need the cert then it will error in the logs.

Would a log warning suffice to address your concern?

Reply:
Update:  I decide to error on invalid cert.  If you specify a cert, most likely you're going to need one.

Reply:
You read my mind there, my concern was to successfully perform the connection without the explicitly configured cert, I think we should rather fail in those cases

Reply:
Good call.